### PR TITLE
Renames check_weakness() on /datum/species to check_species_weakness()

### PR DESCRIPTION
**Pull request by Amelia0010**
So there I was, merging mirror PRs like I do, when I notice this:
image

I think to myself: "why is this not a boolean?" and I look at the /tg/ PR, just to see this:
image

It is at this point that I think that that's incredibly dumb, and I look into how check_weakness works, and find this:
image

Now I get confused, and wonder: "why on god's green earth does it multiply the force with the weakness, if the default weakness is 0?"
The answer to that question is that there's two different check_weakness procs, one on a mob level, and one on a species level. And of course, the mob proc returns the value that it should multiply the force with, whereas the species proc returns the value it should multiply with, minus 1, since the species check_weakness is added on top of the normal check_weakness which just returns the value it should multiply with.

To summarize: 
There's two check_weakness procs, the return result of which could be (and have been) confused with booleans. Not only that, but the return values for the two identically named procs need to be different to achieve the same thing.

Renames check_weakness on the species side to check_species_weakness, since they don't get confused with each other then, and adds a comment to the default return value to make sure they're not confused with booleans in the future.

### CODE REVIEW

File: code/modules/mob/living/carbon/human/human.dm [hunk 1]
```diff
@@ -788,7 +788,7 @@
 /mob/living/carbon/human/check_weakness(obj/item/weapon, mob/living/attacker)
 	. = ..()
 	if (dna && dna.species)
-		. += dna.species.check_weakness(weapon, attacker)
+		. += dna.species.check_species_weakness(weapon, attacker)
```
Comment (monster860) [inline code/modules/mob/living/carbon/human/human.dm]: Nich llease replacw this with a multiply and change the other procs accordingly because using add for this is FUCKING RETARDED

**Comment by monster860** [comment]
Tg coders are fucking retarded.

File: code/modules/mob/living/carbon/human/human.dm (hunk 1 above)
Comment (ShizCalev) [reply code/modules/mob/living/carbon/human/human.dm]: This is correct actually. 

https://github.com/tgstation/tgstation/blob/732b94093525282c2e3dd49f9d864995913a2cf3/code/modules/mob/living/carbon/human/species.dm#L1338-L1339

https://github.com/tgstation/tgstation/blob/f04d6fc8d1bbaa951fa89e9e531a6d256350bff7/code/modules/antagonists/devil/true_devil/_true_devil.dm#L122-L123

**Review by ShizCalev — APPROVED** [review]

File: code/modules/mob/living/carbon/human/human.dm (hunk 1 above)
Comment (nichlas0010) [reply code/modules/mob/living/carbon/human/human.dm]: Right or not, it's dumb :)
